src/amenities/amenities.service.ts

```diff
@@ -10,6 +15,24 @@ import PrismaErrorCodes from 'src/errors/prismaErrorCodes.enum';
 export class AmenitiesService {
   constructor(private prisma: PrismaService) {}
 
+  async verifyOwner(userId: string, accommodationId: string) {
+    try {
+      const accommodation = await this.prisma.accommodation.findUnique({
+        where: {
+          id: accommodationId,
+        },
+        select: {
+          ownerId: true,
+        },
+      });
+      if (!accommodation) throw new NotFoundException('No accomodation found with this id');
+      return userId === accommodation?.ownerId;
+    } catch (error) {
```

Comment:
```
 try {
      const accommodation = await this.prisma.accommodation.findUnique({
        where: {
          id: accommodationId,
          ownerId: userId,
        },
      });
      return accommodation;
    } catch (error) {
      throw new GlobalException(ErrorsTypes.YOUR_ERROR);
    }
```
it is much better, to check exsisting accommodation and owner at once. and remove check for instance in catch block